workspace.go

```diff
@@ -1080,7 +1080,7 @@ func (o WorkspaceCreateOptions) valid() error {
 		o.TriggerPrefixes != nil && len(o.TriggerPrefixes) > 0 {
 		return ErrUnsupportedBothTagsRegexAndTriggerPrefixes
 	}
-	if o.VCSRepo != nil && o.VCSRepo.TagsRegex != nil &&
+	if o.VCSRepo != nil && o.VCSRepo.TagsRegex != nil && *o.VCSRepo.TagsRegex != "" &&
```

Comment:
In looking at this if expression and the similar one when checking the update options, I was thinking it might be clearer to refactor the expression to a helper method that could be called in both places. What do you think?